black_it/loss_functions/msm.py

```diff
@@ -103,12 +107,14 @@ def _validate_covariance_and_calculator(
                 is not 18. Other possible errors won't be caught by this
                 function, and can only be detected at runtime.
         """
-        if covariance_mat is None:
-            # if we were given no covariance matrix, then we'll use a default
-            # one, and we can't do any further validation (without executing the
-            # moment_calculator)
+        if covariance_mat in ("identity", "inverse_variance"):
```

Comment:
Possible improvement. Instead of this, we could have an `Enum` class:

```python
class CovarianceMatrixType:
    IDENTITY = "identity"
    INVERSE_VARIANCE = "inverse_variance"
```

And then here (we should first check it is not a string!):

```python
try:
    covariance_mat = CovarianceMatrixType(covariance_mat)
except ValueError:
    raise ValueError(f"expected one of {list(map(lambda x: x.value, CovarianceMatrixType))}, got {covariance_mat})

```

Reply:
that's a good idea, sooner or later we will implement this for sure as the number of option for the covariance matrix increases, but I will not do it immediately as we effectively have a single possibility now.